includes/payment-methods/class-wc-stripe-express-checkout-helper.php

```diff
@@ -571,85 +571,64 @@ public function is_page_supported() {
 	 * @return  boolean  True if express checkout elements are supported on current page, false otherwise
 	 */
 	public function should_show_express_checkout_button() {
-		// Bail if account is not connected.
-		if ( ! WC_Stripe::get_instance()->connect->is_connected() ) {
-			WC_Stripe_Logger::log( 'Account is not connected.' );
-			return false;
-		}
-
-		// If no SSL bail.
-		if ( ! $this->testmode && ! is_ssl() ) {
-			// phpcs:ignore WordPress.PHP.DevelopmentFunctions
-			WC_Stripe_Logger::log( 'Stripe Express Checkout live mode requires SSL. ' . print_r( [ 'url' => get_permalink() ], true ) );
-			return false;
-		}
+		$should_show_button = false;
 
 		$available_gateways = WC()->payment_gateways->get_available_payment_gateways();
-		if ( ! isset( $available_gateways['stripe'] ) ) {
-			WC_Stripe_Logger::log( 'Stripe Express Checkout requires the Stripe gateway to be enabled.' );
-			return false;
-		}
-
-		// Don't show if on the cart or checkout page, or if page contains the cart or checkout
-		// shortcodes, with items in the cart that aren't supported.
-		if (
-			WC_Stripe_Helper::has_cart_or_checkout_on_current_page()
-			&& ! $this->allowed_items_in_cart()
-		) {
-			WC_Stripe_Logger::log( 'Some items in cart are not compatible with Stripe Express Checkout. ' );
-			return false;
-		}
-
-		// Don't show on cart if disabled.
-		if ( is_cart() && ! $this->should_show_ece_on_cart_page() ) {
-			WC_Stripe_Logger::log( 'Stripe Express Checkout buttons display on cart is disabled. ' );
-			return false;
-		}
-
-		// Don't show on checkout if disabled.
-		if ( is_checkout() && ! $this->should_show_ece_on_checkout_page() ) {
-			WC_Stripe_Logger::log( 'Stripe Express Checkout buttons display on checkout is disabled. ' );
-			return false;
-		}
-
-		// Don't show if product page ECE is disabled.
-		if ( $this->is_product() && ! $this->should_show_ece_on_product_pages() ) {
-			WC_Stripe_Logger::log( 'Stripe Express Checkout buttons display on product pages is disabled. ' );
-			return false;
-		}
-
-		// Don't show if product on current page is not supported.
-		if ( $this->is_product() && ! $this->is_product_supported( $this->get_product() ) ) {
-			WC_Stripe_Logger::log( 'Product is not supported by Stripe Express Checkout. Product ID: ' . $this->get_product()->get_id() );
-			return false;
-		}
-
-		// Don't show if the total price is 0.
-		// ToDo: support free trials. Free trials should be supported if the product does not require shipping.
-		if ( ( ! ( $this->is_pay_for_order_page() || $this->is_product() ) &&
-			isset( WC()->cart ) && ! WC()->cart->is_empty() && 0.0 === (float) WC()->cart->get_total( false ) )
-			|| ( $this->is_product() && 0.0 === (float) $this->get_product()->get_price() )
-		) {
-			WC_Stripe_Logger::log( 'Stripe Express Checkout does not support free products.' );
-			return false;
```

Comment:
Did we lose the "free product" check?

Reply:
Oops 🤦 ! Good catch! Added it back in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4200/commits/6eda9f9a6d214d1639d736a7717f733acb28b956